### PR TITLE
Auth 2.0 Sign-in flow

### PR DESCRIPTION
**Pull request by ivchenkodima**
https://github.com/edenlabllc/ehealth.api/issues/1457
https://github.com/edenlabllc/ehealth.api/issues/1487
https://github.com/edenlabllc/ehealth.api/issues/1485
https://github.com/edenlabllc/ehealth.api/issues/1486

### CODE REVIEW

File: app/common/containers/forms/OtpForm/index.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      submitting,
+      router,
+    } = this.props;
+
+    return (
+      <form className={styles.main} onSubmit={handleSubmit}>
+        <div>
+          <Field
+            placeholder="Введіть код, що прийшов на телефон"
+            // labelText=""
```
Comment (LehaMotovilov) [inline app/common/containers/forms/OtpForm/index.js]: Commented code.

File: app/common/containers/forms/FactorForm/index.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import withStyles from 'withStyles';
+import { withRouter } from 'react-router';
+import { reduxForm, Field } from 'redux-form';
+import { reduxFormValidate } from 'react-nebo15-validate';
+
+import Button, { ButtonsGroup } from 'components/Button';
+import FieldMasked from 'components/reduxForm/FieldMasked';
+
+
```
Comment (LehaMotovilov) [inline app/common/containers/forms/FactorForm/index.js]: Remove line.

File: app/common/containers/pages/RequestFactorPage/index.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import styles from './styles.scss';
+
+@withStyles(styles)
+@connect(null, { onSubmit })
+export default class RequestFactorPage extends React.Component {
+  render() {
+    const {
+      onSubmit = () => {},
+    } = this.props;
```
Comment (LehaMotovilov) [inline app/common/containers/pages/RequestFactorPage/index.js]: ``` const { onSubmit = () => {} } = this.props;  ``` ?

File: app/common/containers/pages/UpdateFactorOtpNewPage/redux.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { push } from 'react-router-redux';
+import { SubmissionError } from 'redux-form';
+import { getLocation } from 'reducers';
+import { approveFactor } from 'redux/factors';
+import { login } from 'redux/session';
+
+export const onSubmit = ({ code }) => (dispatch, getState) =>
+  dispatch(approveFactor(parseInt(code, 10)))
+    .then((action) => {
+      if (action.error) {
```
Comment (LehaMotovilov) [inline app/common/containers/pages/UpdateFactorOtpNewPage/redux.js]: DRY. Same as `UpdateFactorOtpNewPage/redux.js`

File: app/common/redux/factors.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+// import { handleAction } from 'redux-actions';
```
Comment (LehaMotovilov) [inline app/common/redux/factors.js]: Commented code.

File: app/common/redux/factors.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import { AUTH_URL } from 'config';
+import { invoke } from './api';
+
+
```
Comment (LehaMotovilov) [inline app/common/redux/factors.js]: Remove line.

File: app/common/redux/factors.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    ),
+
+  }, 'factor/INIT_FACTOR_FAILURE'],
+  body: {
+    factor: `+380${phone}`,
+    type: 'SMS',
+  },
+}, { auth: true });
+
+
```
Comment (LehaMotovilov) [inline app/common/redux/factors.js]: Remove line.

File: app/common/routes/index.js [hunk 8]
```diff
@@ -13,12 +13,24 @@ import InviteAcceptPage from 'containers/pages/InviteAcceptPage';
...
 import InviteRejectPage from 'containers/pages/InviteRejectPage';
 import SignInPage from 'containers/pages/SignInPage';
+import OtpPage from 'containers/pages/OtpPage';
 import AcceptPage from 'containers/pages/AcceptPage';
 import ConditionPage from 'containers/pages/ConditionPage';
 
+import RequestFactorPage from 'containers/pages/RequestFactorPage';
+import RequestFactorOtpPage from 'containers/pages/RequestFactorOtpPage';
+
+
```
Comment (LehaMotovilov) [inline app/common/routes/index.js]: Remove line.

File: app/common/containers/forms/FactorForm/index.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+import styles from './styles.scss';
+
+@withRouter
+@withStyles(styles)
+@reduxForm({
+  form: 'factor-form',
+  validate: reduxFormValidate({
+    phone: {
+      required: true,
+      phone_number: () => /^\d{9}$/,
```
Comment (LehaMotovilov) [inline app/common/containers/forms/FactorForm/index.js]: We need to discuss this regex.

File: app/common/containers/forms/FactorForm/index.js (hunk 2 above)
Comment (ivchenkodima) [reply app/common/containers/forms/FactorForm/index.js]: +

File: app/common/containers/pages/RequestFactorPage/index.js (hunk 3 above)
Comment (ivchenkodima) [reply app/common/containers/pages/RequestFactorPage/index.js]: +

File: app/common/containers/pages/UpdateFactorOtpNewPage/redux.js (hunk 4 above)
Comment (ivchenkodima) [reply app/common/containers/pages/UpdateFactorOtpNewPage/redux.js]: ?

File: app/common/redux/factors.js (hunk 6 above)
Comment (ivchenkodima) [reply app/common/redux/factors.js]: +

File: app/common/redux/factors.js (hunk 7 above)
Comment (ivchenkodima) [reply app/common/redux/factors.js]: +

File: app/common/routes/index.js (hunk 8 above)
Comment (ivchenkodima) [reply app/common/routes/index.js]: +

File: app/common/containers/pages/RequestFactorPage/index.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+import { onSubmit } from './redux';
+
+import styles from './styles.scss';
+
+@withStyles(styles)
+@connect(null, { onSubmit })
+export default class RequestFactorPage extends React.Component {
+  render() {
+    const {
+      onSubmit = () => {} } = this.props;
```
Comment (LehaMotovilov) [inline app/common/containers/pages/RequestFactorPage/index.js]: Something wrong with linting?

File: app/common/containers/forms/OtpForm/styles.scss [hunk 11]
```diff
@@ -0,0 +1,5 @@
+.main {
```
Comment (LehaMotovilov) [inline app/common/containers/forms/OtpForm/styles.scss]: Same as `app/common/containers/forms/FactorForm/styles.scss`.

File: app/common/containers/pages/RequestFactorOtpPage/styles.scss [hunk 12]
```diff
@@ -0,0 +1,12 @@
+.main {
```
Comment (LehaMotovilov) [inline app/common/containers/pages/RequestFactorOtpPage/styles.scss]: Same as `app/common/containers/pages/OtpPage/styles.scss`.

File: app/common/containers/pages/RequestFactorPage/styles.scss [hunk 13]
```diff
@@ -0,0 +1,12 @@
+.main {
```
Comment (LehaMotovilov) [inline app/common/containers/pages/RequestFactorPage/styles.scss]: Same as `app/common/containers/pages/OtpPage/styles.scss`.

File: app/common/containers/pages/UpdateFactorNewPage/styles.scss [hunk 14]
```diff
@@ -0,0 +1,12 @@
+.main {
```
Comment (LehaMotovilov) [inline app/common/containers/pages/UpdateFactorNewPage/styles.scss]: Same as `app/common/containers/pages/OtpPage/styles.scss`.

File: app/common/containers/pages/UpdateFactorOtpNewPage/styles.scss [hunk 15]
```diff
@@ -0,0 +1,12 @@
+.main {
```
Comment (LehaMotovilov) [inline app/common/containers/pages/UpdateFactorOtpNewPage/styles.scss]: Same as `app/common/containers/pages/OtpPage/styles.scss`.

File: app/common/containers/pages/UpdateFactorSignInPage/styles.scss [hunk 16]
```diff
@@ -0,0 +1,12 @@
+.main {
```
Comment (LehaMotovilov) [inline app/common/containers/pages/UpdateFactorSignInPage/styles.scss]: Same as `app/common/containers/pages/OtpPage/styles.scss`

File: app/common/containers/pages/UpdateFactorSuccessPage/styles.scss [hunk 17]
```diff
@@ -0,0 +1,12 @@
+.main {
```
Comment (LehaMotovilov) [inline app/common/containers/pages/UpdateFactorSuccessPage/styles.scss]: Same as `app/common/containers/pages/OtpPage/styles.scss`

File: app/common/containers/pages/UpdateOldFactorOtpPage/styles.scss [hunk 18]
```diff
@@ -0,0 +1,12 @@
+.main {
```
Comment (LehaMotovilov) [inline app/common/containers/pages/UpdateOldFactorOtpPage/styles.scss]: Same as `app/common/containers/pages/OtpPage/styles.scss`.

File: app/common/routes/index.js [hunk 19]
```diff
@@ -79,10 +89,20 @@ export const configureRoutes = ({ store }) => { // eslint-disable-line
...
             <Route path="sign-in" component={SignInPage} />
+            <Route path="update-factor" component={UpdateFactorSignInPage} />
+
             <Route path="reset" component={ResetPasswordPage} />
             <Route path="reset/:id" component={NewPasswordPage} />
             <Route onEnter={requireAuth}>
               <Route path="accept" component={AcceptPage} />
+              <Route path="otp-send" component={OtpPage} />
+              <Route path="request-factor" component={RequestFactorPage} />
+              <Route path="request-factor/approve" component={RequestFactorOtpPage} />
```
Comment (LehaMotovilov) [inline app/common/routes/index.js]: Mb will be better `RequestFactorApprovePage` ?

File: app/common/routes/index.js [hunk 20]
```diff
@@ -79,10 +89,20 @@ export const configureRoutes = ({ store }) => { // eslint-disable-line
...
+
+              <Route path="update-factor/otp" component={UpdateOldFactorOtpPage} />
+              <Route path="update-factor/phone" component={UpdateFactorNewPage} />
+              <Route path="update-factor/phone/otp" component={UpdateFactorOtpNewPage} />
+              <Route path="update-factor/success" component={UpdateFactorSuccessPage} />
```
Comment (LehaMotovilov) [inline app/common/routes/index.js]: Mb will be better use these names?
```
UpdateFactorOtpPage
UpdateFactorPhonePage
UpdateFactorPhoneOtpPage
```

File: app/common/containers/pages/UpdateFactorOtpNewPage/redux.js (hunk 4 above)
Comment (LehaMotovilov) [reply app/common/containers/pages/UpdateFactorOtpNewPage/redux.js]: `app/common/containers/pages/OtpPage/redux.js`

File: app/common/routes/index.js (hunk 19 above)
Comment (ivchenkodima) [reply app/common/routes/index.js]: +

File: app/common/routes/index.js (hunk 20 above)
Comment (ivchenkodima) [reply app/common/routes/index.js]: +

File: app/common/containers/pages/RequestFactorPage/index.js (hunk 10 above)
Comment (ivchenkodima) [reply app/common/containers/pages/RequestFactorPage/index.js]: +

File: app/common/containers/pages/RequestFactorPage/styles.scss (hunk 13 above)
Comment (ivchenkodima) [reply app/common/containers/pages/RequestFactorPage/styles.scss]: +

File: app/common/containers/forms/OtpForm/styles.scss (hunk 11 above)
Comment (ivchenkodima) [reply app/common/containers/forms/OtpForm/styles.scss]: +

**Review by kiurchv — APPROVED** [review]

**Pull request merged**